src/Elliptic/DiscontinuousGalerkin/DgElementArray.hpp

```diff
@@ -3,24 +3,66 @@
 
 #pragma once
 
+#include <charm++.h>
 #include <cstddef>
 #include <vector>
 
 #include "Domain/Block.hpp"
 #include "Domain/Creators/DomainCreator.hpp"
 #include "Domain/Domain.hpp"
+#include "Domain/ElementDistribution.hpp"
 #include "Domain/OptionTags.hpp"
 #include "Domain/Structure/ElementId.hpp"
 #include "Domain/Structure/InitialElementIds.hpp"
 #include "Domain/Tags.hpp"
 #include "Parallel/Algorithms/AlgorithmArray.hpp"
 #include "Parallel/GlobalCache.hpp"
 #include "Parallel/ParallelComponentHelpers.hpp"
+#include "Parallel/Protocols/ArrayElementsAllocator.hpp"
+#include "Utilities/Numeric.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
 #include "Utilities/System/ParallelInfo.hpp"
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
 
 namespace elliptic {
+
+/// A `Parallel::protocols::ArrayElementsAllocator` that creates array elements
```

Comment:
I feel like this documentation is missing the key information on _how_ the domain is covered, no?